app/models/post.rb

```diff
@@ -1,5 +1,13 @@
 class Post < ApplicationRecord
   belongs_to :user
   has_many :comments
+  
+  before_destroy :delete_associated_comments
```

Comment:
I think previously you had a dependent: destroy line in this model which should have taken care of this process for you whenever a post was deleted since the comments would be associated with the post. Unless there is something that I'm missing.